bottlecap/src/lifecycle/flush_control.rs

```diff
@@ -1,14 +1,11 @@
 use crate::config::flush_strategy::FlushStrategy;
-use crate::lifecycle::invocation_times::InvocationTimes;
-use ::std::time;
-use tracing::debug;
+use tokio::time::Interval;
 
-const TWENTY_SECONDS: u64 = 20 * 1000;
+const DEFAULT_FLUSH_INTERVAL: u64 = 1000; // 1s
 
+#[derive(Clone, Copy, Debug, PartialEq)]
```

Comment:
Do we need all of that?
Maybe with an enum rather than struct it could be easier, instead of wrapping the enum on a struct. Also there won't be need of all those ifs later in the logic

Reply:
Clone + Copy the compiler was complaining about, debug and partial eq for debugging

Reply:
I also suspect we might need to pass timeout racing + signals for buffers, that's why I didn't remove the structure yet

Reply:
an enum should be able to get arguments, already implemented those traits and be enough for this use case

Reply:
If I do it, then initialization would end up with something like
```rust
let flush_control = match config.serverless_flush_strategy {
    FlushStrategy::Default => FlushControl::Default,
    FlushStrategy::Periodically(p) => FlushControl::Periodically { interval: p.interval },
    FlushStrategy::EndPeriodically(p) => FlushControl::EndPeriodically { interval: p.interval },
    FlushStrategy::End => FlushControl::End,
};```

Reply:
Will refactor in another PR